### PR TITLE
Issue 262 card restyling

### PR DESCRIPTION
**Pull request by jeka-leb**
develop

## GitHub Board

https://github.com/ita-social-projects/dokazovi-fe/issues/262

#262 single card restyling 

## Summary of issue

- [x]  restyle PostPreviewCard component in accordance with the design

## Summary of change

- [x]  PostPreviewCard restyled
- [x] PostPreviewCardStyles clarifed
- [x] PostPreviewCard.spec.tsx updated
- [x]  theme.ts clarified 
- [x] new fonts connected

### CODE REVIEW

File: src/lib/theme/typography.ts [hunk 1]
```diff
@@ -48,4 +67,18 @@ export const TYPOGRAPHY = {
     fontSize: '14px',
     lineHeight: '18px',
   },
+  subtitle1: {
```
Comment (SergeOlabin) [inline src/lib/theme/typography.ts:70]: can it be more descriptive? 
e.g. postCardSubtitle or something.. subtitle1 | subtitle2 is not good, at some point you'd have subtitle22 and how to distinguish them ..

File: public/index.html [hunk 2]
```diff
@@ -1,21 +1,19 @@
 <!DOCTYPE html>
```
Comment (SergeOlabin) [inline public/index.html:1]: pls follow the identation in this file

File: src/lib/components/PostPreview/PostPreviewCard.tsx [hunk 3]
```diff
@@ -1,13 +1,14 @@
...
+import { ThemeProvider } from '@material-ui/core/styles';
 import Card from '@material-ui/core/Card';
 import Box from '@material-ui/core/Box';
 import CardContent from '@material-ui/core/CardContent';
 import CardMedia from '@material-ui/core/CardMedia';
 import Typography from '@material-ui/core/Typography';
 import { IPost } from '../../types';
 import { useStyles } from '../../styles/PostPreviewCard.styles';
-import PostDirectionLink from '../PostDirectionLink';
+import { MAIN_THEME as mainTheme } from '../../theme/theme';
```
Comment (SergeOlabin) [inline src/lib/components/PostPreview/PostPreviewCard.tsx:11]: just use as MAIN_THEME

**Review by SergeOlabin — CHANGES_REQUESTED** [review]

File: src/lib/components/PostPreview/PostPreviewCard.tsx [hunk 4]
```diff
@@ -1,129 +1,135 @@
...
   const { data } = props;
-  const history = useHistory();
-
+  let expertLink = ``;
+  const postLink = `/posts/${data.id}`;
   let authorFullName = '';
+
   if (data.author?.firstName && data.author?.lastName) {
     authorFullName = `${data.author?.firstName} ${data.author?.lastName}`;
+    expertLink = `/experts/${data.author.id}`;
```
Comment (SergeOlabin) [inline src/lib/components/PostPreview/PostPreviewCard.tsx:24]: technically this doesn't depend on if check above, so it's correct not to put here

File: src/lib/components/PostPreview/PostPreviewCard.tsx [hunk 5]
```diff
@@ -1,129 +1,135 @@
...
   const classes = useStyles();
   return (
-    <Card className={classes.root}>
-      <Box
-        display="flex"
-        flexDirection="column"
-        justifyContent="space-between"
-        height="100%"
-      >
+    <ThemeProvider theme={MAIN_THEME}>
```
Comment (SergeOlabin) [inline src/lib/components/PostPreview/PostPreviewCard.tsx:29]: only one theme provider app.tsx should be enough, pls double check

File: src/lib/styles/PostPreviewCard.styles.ts [hunk 6]
```diff
@@ -1,27 +1,39 @@
...
     body: {
-      margin: '15px',
+      margin: `${theme.spacing(1)}px ${theme.spacing(4)}px ${theme.spacing(
+        5,
+      )}px`,
       cursor: 'pointer',
     },
-  },
+    postType: {
+      color: '#3B6F95',
```
Comment (SergeOlabin) [inline src/lib/styles/PostPreviewCard.styles.ts:27]: pls use theme.palette

**Review by SergeOlabin — CHANGES_REQUESTED** [review]

File: src/lib/theme/typography.ts (hunk 1 above)
Comment (jeka-leb) [reply src/lib/theme/typography.ts:70]: As of today, Material UI library V4, which is currently used on the project, provides us with the limited number of Typography default  variants names.

File: public/index.html (hunk 2 above)
Comment (jeka-leb) [reply public/index.html]: agreed and fixed

File: src/lib/components/PostPreview/PostPreviewCard.tsx (hunk 3 above)
Comment (jeka-leb) [reply src/lib/components/PostPreview/PostPreviewCard.tsx]: agreed and fixed

File: src/lib/styles/PostPreviewCard.styles.ts [hunk 7]
```diff
@@ -4,21 +4,21 @@ export const useStyles = makeStyles(
...
     },
     header: {
       display: 'flex',
-      margin: '15px 15px 0 15px',
+      margin: '0 22px',
       cursor: 'pointer',
     },
     body: {
-      margin: '15px',
+      margin: '5px 22px 25px',
```
[the remark this answers is not in the document]
Comment (jeka-leb) [reply src/lib/styles/PostPreviewCard.styles.ts]: agreed, corresponding files updated in accordance with the notes

File: src/lib/components/PostPreview/PostPreviewCard.tsx [hunk 8]
```diff
@@ -34,96 +35,108 @@ const PostPreviewCard: React.FC<IPostPreviewCardProps> = (props) => {
...
-                {data.postType.name}
+            <Box
+              className={classes.body}
+              display="flex"
+              flexDirection="column"
+              flexGrow="1"
+              justifyContent="space-between"
+              onClick={goPostView}
+            >
+              <CardContent style={{ padding: '0' }}>
```
[the remark this answers is not in the document]
Comment (jeka-leb) [reply src/lib/components/PostPreview/PostPreviewCard.tsx]: agreed and fixed

File: src/lib/components/PostPreview/PostPreviewCard.tsx [hunk 9]
```diff
@@ -34,96 +35,108 @@ const PostPreviewCard: React.FC<IPostPreviewCardProps> = (props) => {
...
-                {authorFullName}
-              </Typography>
+                <Box
+                  display="flex"
+                  flexDirection="column"
+                  mt={2}
+                  justifyContent="space-between"
+                >
+                  <Typography
+                    style={{ color: '#3B6F95' }}
```
[the remark this answers is not in the document]
Comment (jeka-leb) [reply src/lib/components/PostPreview/PostPreviewCard.tsx]: the color used from palette

File: src/lib/components/PostPreview/PostPreviewCard.tsx [hunk 10]
```diff
@@ -34,96 +35,108 @@ const PostPreviewCard: React.FC<IPostPreviewCardProps> = (props) => {
...
+                    variant="subtitle1"
+                    component="h3"
+                  >
+                    {data.postType.name.toUpperCase()}
+                  </Typography>
+                  <Typography
+                    align="left"
+                    variant="h4"
+                    component="h4"
+                    style={{ fontSize: '16px', lineHeight: '19px' }}
```
[the remark this answers is not in the document]
Comment (jeka-leb) [reply src/lib/components/PostPreview/PostPreviewCard.tsx]: agreed and fixed

File: src/lib/components/PostPreview/PostPreviewCard.tsx (hunk 5 above)
Comment (jeka-leb) [reply src/lib/components/PostPreview/PostPreviewCard.tsx:29]: repeated ThemeProvider components removed

File: src/lib/styles/PostPreviewCard.styles.ts (hunk 6 above)
Comment (jeka-leb) [reply src/lib/styles/PostPreviewCard.styles.ts:27]: agreed, theme.palette  is used

File: src/lib/components/PostPreview/PostPreviewCard.tsx (hunk 4 above)
Comment (jeka-leb) [reply src/lib/components/PostPreview/PostPreviewCard.tsx]: modified utilities/types.ts, interface IPost { author: IExpert}, so author is not already optional field. Therefore mentioned checks are no more required

**Review by darya42185 — APPROVED** [review]

File: src/lib/types.ts [hunk 11]
```diff
@@ -39,7 +39,7 @@ export interface IPost {
   id: number;
   title: string;
   content?: string;
-  author?: IExpert;
+  author: IExpert;
```
Comment (SergeOlabin) [inline src/lib/types.ts:42]: 👍

File: src/modules/posts/components/PostView.tsx [hunk 12]
```diff
@@ -12,10 +12,10 @@ export interface IPostViewProps {
...
   const classes = useStyles();
 
-  let authorFullName = '';
-  if (post.author?.firstName && post.author?.lastName) {
-    authorFullName = `${post.author?.firstName} ${post.author?.lastName}`;
-  }
+  const authorFullName = `${post.author.firstName} ${post.author.lastName}`;
+  // if (post.author?.firstName && post.author?.lastName) {
+  //   authorFullName = `${post.author?.firstName} ${post.author?.lastName}`;
+  // }
```
Comment (SergeOlabin) [inline src/modules/posts/components/PostView.tsx:18]: remove :)

**Review by SergeOlabin — CHANGES_REQUESTED** [review]

File: src/lib/styles/PostPreviewCard.styles.ts [hunk 13]
```diff
@@ -1,27 +1,51 @@
...
+    authorTypography: {
+      fontSize: '16px',
+      lineHeight: '19px',
+    },
+    content: {
+      padding: 0,
+    },
+    eyeNumber: {
+      color: theme.palette.text.secondary,
+      marginLeft: '6px',
```
Comment (SergeOlabin) [inline src/lib/styles/PostPreviewCard.styles.ts:38]: pls use spacing

File: src/lib/styles/PostPreviewCard.styles.ts [hunk 14]
```diff
@@ -1,27 +1,51 @@
...
+      fontSize: '9px',
+      lineHeight: '11px',
+    },
+    eyeIcon: {
+      width: '23px',
+      height: '13px',
+      marginLeft: '-3px',
```
Comment (SergeOlabin) [inline src/lib/styles/PostPreviewCard.styles.ts:45]: pls use spacing

File: src/lib/styles/PostPreviewCard.styles.ts [hunk 15]
```diff
@@ -1,27 +1,51 @@
...
+      color: '#6D6E71',
```
Comment (SergeOlabin) [inline src/lib/styles/PostPreviewCard.styles.ts:46]: pls use palette

**Review by SergeOlabin — CHANGES_REQUESTED** [review]

File: src/modules/posts/components/PostView.tsx (hunk 12 above)
Comment (jeka-leb) [reply src/modules/posts/components/PostView.tsx]: comments removed ;)

File: src/lib/styles/PostPreviewCard.styles.ts (hunk 13 above)
Comment (jeka-leb) [reply src/lib/styles/PostPreviewCard.styles.ts]: fixed

File: src/lib/styles/PostPreviewCard.styles.ts (hunk 15 above)
Comment (jeka-leb) [reply src/lib/styles/PostPreviewCard.styles.ts]: set a new property "secondary" in the palette.ts for this purpose

File: src/lib/styles/PostPreviewCard.styles.ts (hunk 14 above)
Comment (jeka-leb) [reply src/lib/styles/PostPreviewCard.styles.ts]: fixed

**Review by SergeOlabin — APPROVED** [review]

**Pull request merged**